src/main/java/org/spongepowered/common/mixin/core/network/NetHandlerPlayServerMixin.java

```diff
@@ -396,73 +397,49 @@ private boolean throwMoveEvent(final EntityPlayerMP playerMP, final CPacketPlaye
 
-            // Minecraft sends a 0, 0, 0 position when rotation only update occurs, this needs to be recognized and corrected
-            boolean rotationOnly = !packetIn.moving && packetIn.rotating;
-
-            if (rotationOnly) {
-                // Correct the to location so it's not misrepresented to plugins, only when player rotates without moving
-                // In this case it's only a rotation update, which isn't related to the to location
-                fromLocation = player.getLocation();
-                toLocation = fromLocation;
+            final double deltaSquared = toPosition.distanceSquared(fromPosition);
 
-                rotationOnly = ShouldFire.ROTATE_ENTITY_EVENT;
-            }
 
-            if (packetIn.moving && packetIn.rotating) {
-                positionOnly = !toLocation.getPosition().equals(fromLocation.getPosition()) && ShouldFire.MOVE_ENTITY_EVENT_POSITION;
-                rotationOnly = ShouldFire.ROTATE_ENTITY_EVENT;
-            }
-
-            mixinPlayer.bridge$setVelocityOverride(toLocation.getPosition().sub(fromLocation.getPosition()));
+            final Vector3d fromRotation = player.getRotation();
+            final Vector3d toRotation = packetIn.rotating ? Vector3d.from(packetIn.pitch, packetIn.yaw, 0) : fromRotation;
 
-            final double deltaSquared = toLocation.getPosition().distanceSquared(fromLocation.getPosition());
             final double deltaAngleSquared = fromRotation.distanceSquared(toRotation);
 
+            mixinPlayer.bridge$setVelocityOverride(toPosition.sub(fromPosition));
+
             // These magic numbers are sad but help prevent excessive lag from this event.
             // eventually it would be nice to not have them
-            if (deltaSquared > ((1f / 16) * (1f / 16)) || deltaAngleSquared > (.15f * .15f)) {
-                final Transform<World> fromTransform = player.getTransform().setLocation(fromLocation).setRotation(fromRotation);
-                Transform<World> toTransform = player.getTransform().setLocation(toLocation).setRotation(toRotation);
+            final boolean moving = deltaSquared > ((1f / 16) * (1f / 16));
+            final boolean rotating = deltaAngleSquared > (.15f * .15f);
+            if (moving || rotating) {
+                final Transform<World> fromTransform = player.getTransform().setPosition(fromPosition).setRotation(fromRotation);
+                Transform<World> toTransform = player.getTransform().setPosition(toPosition).setRotation(toRotation);
                 final Transform<World> originalToTransform = toTransform;
-                if (rotationOnly || positionOnly) {
-                    final Event event;
-                    if (rotationOnly) {
-                        event = SpongeEventFactory.createRotateEntityEvent(Sponge.getCauseStackManager().getCurrentCause(), fromTransform, toTransform, player);
+                if (rotating && ShouldFire.ROTATE_ENTITY_EVENT && !packetIn.moving) {
```

Comment:
This introduces a bug. It should not fire if `moving` is `false` rather than this - you can get this packet but it not be in the tolerances. What this now does is not fire an event if `rotating` is `true` (rotation is large enough) but `moving` is `false` - if rotation is true but moving is false, then this should still fire.